### PR TITLE
Task 6

### PR DESCRIPTION
**Pull request by Savsan**
FE link for testing: https://d24frtl7t8xxi1.cloudfront.net/admin/products

- contains configuration for catalogBatchProcess function - **DONE**
- contains policies to allow lambda catalogBatchProcess function to interact with SNS and SQS - **DONE**
- contains configuration for SQS catalogItemsQueue - **DONE**
- contains configuration for SNS Topic createProductTopic and email subscription - **DONE**
- catalogBatchProcess lambda is covered by unit tests - **NOT DONE**
- set a Filter Policy for SNS createProductTopic - **DONE**

### CODE REVIEW

File: product-service/serverless.ts [hunk 1]
```diff
@@ -37,6 +37,73 @@ const serverlessConfiguration: Serverless = {
...
+      importProductSnsTopic: {
+        Type: 'AWS::SNS::Topic',
+        Properties: {
+          TopicName: `${process.env.IMPORT_PRODUCT_SNS_TOPIC_NAME}-${process.env.STAGE}`,
+        }
+      },
+      importProductSnsSubscription: {
+        Type: 'AWS::SNS::Subscription',
+        Properties: {
+          Endpoint: 'vorobev.profi@gmail.com',
```
Comment (evgenii-alekseenko) [inline product-service/serverless.ts:79]: This is sensitive information, therefore it's better to put it to env vars

File: product-service/.env [hunk 2]
```diff
@@ -5,3 +5,5 @@ DB_PORT=
 DB_NAME=
```
Comment (evgenii-alekseenko) [inline product-service/.env:5]: Just for interest. Usually `.env` file ignored by `.gitignore`. How do you deploy? Do you hav some other `***.env` file for deployment or do you manually put proper credentials here?

File: product-service/handlers/catalogBatchProcess.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+        await publishToImportProductsSnsTopic({
+            sns,
+            products: addedProducts,
+            status: 'SUCCESS',
+        });
+    } catch (error) {
+        logErrorRelatedData({ event, error });
+
+        return {
```
Comment (evgenii-alekseenko) [inline product-service/handlers/catalogBatchProcess.ts:47]: As I understand lambda `catalogBatchProcess` should have `SQSHandler` type, it's triggered by SQS queue.
So, you don't need to return anything from here.
image

**Review by evgenii-alekseenko — APPROVED** [review]
The app works. Good job!

File: product-service/db/DBClient.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    createAddProductQuery,
+    createAddStockQuery,
+    createGetProductByIdQuery,
+    createGetProductListQuery,
+} from './queries';
+import { BEGIN_DB_TRANSACTION, COMMIT_DB_TRANSACTION, ROLLBACK_DB_TRANSACTION } from './constants';
+
+import { IAddProductBody } from './types';
+
+export default class DBClient {
```
Comment (Panibrat) [inline product-service/db/DBClient.ts:13]: I guess you can import and use db client from product-service

File: product-service/handler.ts [hunk 5]
```diff
@@ -1,5 +1,6 @@
 import { getProductsList } from './handlers/getProductsList';
```
Comment (Panibrat) [inline product-service/handler.ts:1]: export * from './handlers/getProductsList';

File: product-service/handlers/catalogBatchProcess.ts (hunk 3 above)
Comment (Panibrat) [reply product-service/handlers/catalogBatchProcess.ts:47]: +1

File: product-service/serverless.ts (hunk 1 above)
Comment (Panibrat) [reply product-service/serverless.ts:79]: +1

**Pull request merged**